### PR TITLE
Improve action items

### PR DESCRIPTION
**Pull request by mdmohsin7**
- [x] Improve prompt for action items
- [x] Check/Uncheck action items
- [x] Delete action items
- [x] Analytics to track

https://github.com/user-attachments/assets/8a4d3918-eda1-4088-ab29-85f4b3e52c86

## Release Notes

- **New Features**
  - Users can now update the state of action items and delete them directly from their memories.
  - Added functionality to restore deleted action items or permanently remove them.
  - Introduced a new widget for displaying and managing action items more effectively, including swipe-to-delete and undo options.

- **Bug Fixes**
  - Enhanced error handling for authentication and API requests.

- **Improvements**
  - Improved tracking of user interactions with action items for better analytics.
  - Refined prompts for AI interactions to enhance clarity and efficiency.
  - Enhanced handling of HTTP DELETE requests to include request bodies.
  - Added a new property to track the deletion state of action items.

### CODE REVIEW

File: backend/utils/llm.py [hunk 1]
```diff
@@ -74,12 +74,12 @@ def should_discard_memory(transcript: str) -> bool:
 def get_transcript_structure(transcript: str, started_at: datetime, language_code: str, tz: str) -> Structured:
     prompt = ChatPromptTemplate.from_messages([(
         'system',
-        '''Your task is to provide structure and clarity to the recording transcription of a conversation.
+        '''You are an expert conversation analyzer. Your task is to analyze the conversation and provide structure and clarity to the recording transcription of a conversation.
```
Comment (beastoin) [inline backend/utils/llm.py:77]: why we need to update the prompt?

**Review by beastoin — COMMENTED** [review]
should we have a better UI for the Action Item > checkbox sir ?

**Comment by mdmohsin7** [comment]
> could we have a better UI for the Action Item > checkbox sir ?

Does making it a bit rounded works? Or complete circle maybe?

File: backend/utils/llm.py (hunk 1 above)
Comment (mdmohsin7) [reply backend/utils/llm.py:77]: The previous prompt would also include a lot of items that the user might have already done, and it sometimes also includes items that are not necessarily actionable. The updated prompt forces it to make sure that the items are concise, actionable and are something that are not already completed by the user

Comment (mdmohsin7) [reply backend/utils/llm.py:77]: Ref:
https://github.com/BasedHardware/omi/issues/1123#issuecomment-2429615914

Comment (mdmohsin7) [reply backend/utils/llm.py:77]: Oh you are referring to the `You are an expert conversation analyzer.` part specifically, it is to improve the quality of things it extracts from the transcript. GPT tends to do better when given a specific role to it

**Comment by mdmohsin7** [comment]
IMG_BE6BE310DC67-1

**Comment by beastoin** [comment]
image


try circle pls sir, just likes the current checkbox.

**Comment by mdmohsin7** [comment]
> try circle pls sir, just likes the current checkbox.

Alright let me

Btw I can see the next task you are planning to assign 👀

**Comment by mdmohsin7** [comment]
IMG_59B64E69711C-1

**Comment by beastoin** [comment]
lgtm @mdmohsin7 🚀

**Pull request merged**